client/src/schema/repository.ts

```diff
@@ -352,17 +356,26 @@ export class BaseClient<D extends Record<string, Repository<any>> = Record<strin
       const [field, linkTable] = link;
       const value = o[field];
 
-      if (value && typeof value === 'object') {
-        const { id } = value as any;
-        if (Object.keys(value).find((col) => col === 'id')) {
-          o[field] = this.initObject(linkTable, value);
-        } else if (id) {
+      if (isIdentifiable(value)) {
+        const { id } = value;
+
+        if (depth === -1 || depth >= 10) {
           o[field] = {
             id,
-            get: () => {
-              this.db[linkTable].read(id);
+            xata: { version: 1 },
+            read: () => {
+              return this.db[linkTable].read(id);
+            },
+            update: (data: any) => {
+              return this.db[linkTable].update(id, data);
+            },
+            delete: () => {
+              return this.db[linkTable].delete(id);
             }
           };
+        } else if (id) {
+          const item = await this.db[linkTable].read(id, { ignoreLinks: true });
```

Comment:
This is making a request to the server, right? This would cause an N+1 problem. We should just `initObject` but not make a request to the server. Even if we just have the id.

Reply:
It will make up to 10 requests (max depth).

If we only initObject, all properties except ``id`` will always be undefined (which is not what we want)

Reply:
You can take a look at the integration test I have added (has an infinite recursive depedency) that ensures we don't have N + 1 calls here.